src/NServiceBus.Gateway/Settings/GatewaySettings.cs

```diff
@@ -135,6 +135,7 @@ public void AddReceiveChannel(string address, string type = "http", int maxConcu
             if (settings.TryGet(out List<ReceiveChannel> channels))
             {
                 channels.Add(channel);
+                return;
```

Comment:
We should simplify the logic in this method by using `settings.GetOrCreate` instead.